src/dtproject/types.py

```diff
@@ -158,5 +159,28 @@ class DevContainerConfiguration(dict):
     workspaceFolder: Optional[str] = None
 
 
+@dataclasses.dataclass
+class Hook:
+    command: str
+    required: Optional[bool] = False
+
+
+@dataclasses.dataclass
+class LayerHooks(DataClassLayer):
+    hooks: Dict[str, List[Hook]] = dataclasses.field(default_factory=lambda: ddict(list))
+
+    def __post_init__(self):
+        self.hooks = ddict(list, self.hooks)
+
```

Comment:
@afdaniele I had to add this to instantiate `hooks` as a `ddict` when we load it from a yaml file, otherwise it would be loaded as a `dict` and throw a `KeyError` when we tried to access non-existing keys.